### DISCUSSION TITLE
Thumbnails randomly not showing

### DISCUSSION THREAD

**Discussion by gogo199432** [photoprism/photoprism#1835] [Q&A] [answered]
Hey, did anyone encounter this issue? Have originally posted as an issue, but was directed to open a thread here instead

 #1647

**Comment by lastzero** [comment]
We've just added a special checklist for broken or missing thumbnails:
https://docs.photoprism.app/getting-started/troubleshooting/#broken-thumbnails

**Reply by gogo199432** [reply]
Okay so it looks like the issue is somehow with my proxy setup cause after connecting to it by local port the issue disappears. However I'm unsure what the issue could be, since I tried only with using my redirect middleware and the issue persisted. However the config should be identical to the ones in the docs? Could you / someone take a look please? (I have removed most of the config to make it as small as possible)

[traefik.yml](https://pastebin.com/EuvSfGMc)
[fileprovider](https://pastebin.com/ySbJmj7h)
[photoprism docker](https://pastebin.com/XZYi8WsZ)

**Reply by gogo199432** [reply]
Update: It turns out I'm getting 521 errors for those thumbnails that do not load, so I guess there is some weird issue going on with cloudflare. Question is what though, because it decides seemingly at random when it wants to work for a given picture.

**Comment by gogo199432** [comment, marked as answer]
For anyone stumbling onto this in the future. For me the issue was that when I originally setup my OpenWRT router, I have put a rate limiting on the traefik port mappings. I guess the reason this became an issue only now is that photoprism needs to make a lot of requests within a short amount of time when loading the pictures so they were being dropped.
